R/getStepwiseDivergence.R

```diff
@@ -222,3 +233,36 @@ setMethod("getTimeDivergence",
     return(x)
 
 }
+
+############################ HELPER FUNCTION(S) ################################
+
+.get_mat_from_se <- function (x, assay_name, altexp) {
+    # non-empty string
+    if(!.is_non_empty_string(assay_name)){
+        stop("'assay_name' must be a non-empty single character value.",
+             call. = FALSE)
+    }
+    if (!is.null(altexp)) {
+        x <- altExp(x, altexp)
+    }
+    mat <- NULL
+    if (assay_name %in% assayNames(x)) {
+        mat <- t(assay(x, assay_name))
+    }
+    # x has to be an instance of SingleCellExperiment to have a reducedDim slot
+    if (is(x, "SingleCellExperiment") && assay_name %in% reducedDimNames(x)) {
+        if (!is.null(mat))
+            # if present in both assay and reducedDim
+            warning("'assay_name' was present in assay and reducedDim. ",
+                    "'assay_name' is picked from assay.",
```

Comment:
I would check immediately at the start of the script to avoid any unnecessary further steps.

If available in both -> use assay, provide warning
If available in just one -> pick accordingly

There could be helper function for picking each so that it is nicely splitted in components.